### PR TITLE
OAuth database connectors

### LINKED ISSUE #7: Use OAuth to authenticate to bigquery
It would be great to use OAuth to authenticate to BQ. You could expand the user base that way without having to rely on someones ability to create a service account

[Details](https://cloud.google.com/bigquery/docs/authorization)

### PR DESCRIPTION
**Pull request by csjh**
### Description

Resolves #7, #96

Adds support for BigQuery authentication through OAuth access token & gcloud CLI, and Snowflake authentication through key/pair, okta, & browser-based SSO

### Checklist

- [x] I have added a [changeset](https://github.com/evidence-dev/evidence/blob/main/CONTRIBUTING.md#adding-a-changeset)

### CODE REVIEW

**Comment by csjh** [comment]
Realized the raw OAuth option for snowflake is a bit redundant since it'll always be used on the client (which `externalbrowser` does anyways). Might be worth revisiting in the future as an option for cloud deployments.

File: packages/bigquery/index.cjs [hunk 1]
```diff
@@ -30,45 +31,66 @@ const standardizeResult = async (result) => {
...
-				}
-			};
-			return creds;
-		}
-	} catch {
-		throw new Error('Missing database credentials');
+const getCredentials = (database = {}) => {
+	const authentication_method =
+		database.authenticator ??
+		process.env['BIGQUERY_AUTHENTICATOR'] ??
```
Comment (ItsMeBrianD) [inline packages/bigquery/index.cjs:37]: This is going to have merge conflicts when the environment variables PR is merged.

File: sites/example-project/src/components/ui/Databases/GenericForm.svelte [hunk 2]
```diff
@@ -5,9 +5,9 @@
 	export let credentials;
 	export let disableSave;
 
-	let requiredOpts = opts.filter((d) => d.optional !== true);
-	let optionalOpts = opts.filter((d) => d.optional === true);
-	let overrideOpts = opts.filter((d) => d.optional === true && d.override === true);
+	$: requiredOpts = opts.filter((d) => d.optional !== true);
+	$: optionalOpts = opts.filter((d) => d.optional === true);
+	$: overrideOpts = opts.filter((d) => d.optional === true && d.override === true);
```
Comment (ItsMeBrianD) [inline sites/example-project/src/components/ui/Databases/GenericForm.svelte:11]: These variables should still be declared (e.g. with `let`)

Comment (csjh) [reply sites/example-project/src/components/ui/Databases/GenericForm.svelte:11]: Why? The compiler inserts declarations for reactively declared variables.

Comment (ItsMeBrianD) [reply sites/example-project/src/components/ui/Databases/GenericForm.svelte:11]: Oh that's neat I didn't know that

**Comment by archiewood** [comment]
@csjh I'd love a demo of how to use the snowflake browser auth at some point. 
When I pulled the branch I was struggling to do it.

**Comment by archiewood** [comment]
I also have some UI nits - apologies - I didn't realize you were so close to merging or I would have commented them earlier. I can open a new issue if that's easiest? Or reopen the existing one?

**Comment by csjh** [comment]
For the browser auth you have to set up an SSO provider on your Snowflake account (in my testing I used Okta). Could revise the docs to make that clearer. 

And yeah might as well reopen the same ticket.

**Comment by archiewood** [comment]
Ah okay - so to clarify - you cannot use the snowflake login with this implementation?
image

**Comment by csjh** [comment]
Yeah, though I just noticed that the errors aren't very helpful to realizing that on your own, I'll fix that up with your UI changes

**Pull request merged**